src/v/cloud_storage/segment_meta_cstore.cc

```diff
@@ -1021,6 +1112,31 @@ class segment_meta_cstore::impl
         if (_write_buffer.size() > max_buffer_entries) {
             flush_write_buffer();
         }
+        return true;
+    }
+
+    auto covered_range(const segment_meta& sm) const -> std::pair<
```

Comment:
looks like this should go into a separate commit

Reply:
done